packages/medusa/src/subscribers/search-indexing.ts

```diff
@@ -0,0 +1,94 @@
+import EventBusService from "../services/event-bus"
+import { SEARCH_INDEX_EVENT } from "../loaders/search-index"
+import ProductService from "../services/product"
+import { indexTypes } from "medusa-core-utils"
+import { SearchService } from "../services"
+import { Product } from "../models"
+
+type InjectedDependencies = {
+  eventBusService: EventBusService
+  searchService: SearchService
+  productService: ProductService
+}
+
+class SearchIndexingSubscriber {
+  private readonly eventBusService_: EventBusService
+  private readonly searchService_: SearchService
+  private readonly productService_: ProductService
+
+  constructor({
+    eventBusService,
+    searchService,
+    productService,
+  }: InjectedDependencies) {
+    this.eventBusService_ = eventBusService
+    this.searchService_ = searchService
+    this.productService_ = productService
+
+    this.eventBusService_.subscribe(SEARCH_INDEX_EVENT, this.indexDocuments)
+  }
+
+  indexDocuments = async (): Promise<void> => {
+    const TAKE = 1000
```

Comment:
suggestion, very-nit(non-blocking): We could make the batch size configurable through plugin options as suggested in this [issue](https://github.com/medusajs/medusa/issues/1393). Was initially suggested by the meilisearch team

Reply:
ok I ve added it, but in order to not update to much the existing, the options will only exists if the the search service from a plugin set the options on the class properties

Reply:
Yeah, makes sense! Though I am not sure, that we need the `async options()` method do we? @adrien2p

Reply:
What async method are you talking about @olivermrbl

Reply:
also, I added a getter to ensure that you can "only" get the options but not mutating them directly on the class (even if in javascript you can access it, but at least the IDE will raise an error)

Reply:
Yeah, was referring to the getter. Did not mean to include the `async` 😅

Reply:
I added a getter, because it is a private property and we should not get access to it and allowed to modify it. The getter only expose it as a readonly value

Reply:
Got it - I like it :)